pyfstat/grid_based_searches.py

```diff
@@ -271,48 +276,51 @@ def run(self, return_data=False):
                 return
 
         if hasattr(self, "search") is False:
-            self.inititate_search_object()
+            self._initiate_search_object()
 
-        data = []
-        for vals in tqdm(iterable, total=getattr(self, "total_iterations", None)):
+        output_dtype = np.dtype(
+            {
+                "names": self.output_keys,
+                "formats": np.repeat(float, len(self.output_keys)),
+            }
+        )
+        data = np.zeros(len(self.input_data), dtype=output_dtype)
+        for n, vals in enumerate(
+            tqdm(iterable, total=getattr(self, "total_iterations", None))
+        ):
             detstat = self.search.get_det_stat(*vals)
             thisCand = list(vals) + [detstat]
-            data.append(thisCand)
+            for k, key in enumerate(self.output_keys):
+                data[key][n] = thisCand[k]
```

Comment:
I didn't find a nice one-liner for this due to the general clunkiness of numpy structured arrays. Maybe we want to check timing on this.

Reply:
Not sure of what's going on here either. Why do you use `np.dtype`?

Reply:
I think mostly to work nicely with `np.savetxt` and `np.genfromtxt`. Without setting the `dtype` manually, an `np.array` with named columns would always have them all as floats.

Reply:
Testing and researching a bit more: and the specific reason for why I explicitly set the `formats` (i.e. actual data type) of the `dtype` as well is that I can't give it `names` alone, nor define a plain `np.array` and later change its `data.dtype.names` (-> "ValueError: there are no fields defined").

Alternatives would be to use `pandas`, or simple dicts of arrays (which would be more analogous to the MCMC class, so maybe actually a good idea!).